osu.Game/Skinning/Skin.cs

```diff
@@ -113,18 +114,37 @@ protected Skin(SkinInfo skin, IStorageResourceProvider? resources, IResourceStor
                 {
                     string jsonContent = Encoding.UTF8.GetString(bytes);
 
-                    // handle namespace changes...
+                    SkinLayoutInfo? layoutInfo = null;
 
-                    // can be removed 2023-01-31
-                    jsonContent = jsonContent.Replace(@"osu.Game.Screens.Play.SongProgress", @"osu.Game.Screens.Play.HUD.DefaultSongProgress");
-                    jsonContent = jsonContent.Replace(@"osu.Game.Screens.Play.HUD.LegacyComboCounter", @"osu.Game.Skinning.LegacyComboCounter");
+                    try
+                    {
+                        // First attempt to deserialise using the new SkinLayoutInfo format
+                        layoutInfo = JsonConvert.DeserializeObject<SkinLayoutInfo>(jsonContent);
+                    }
+                    catch
+                    {
+                    }
+
+                    // if that fails, attempt to deserialise using the old naked list.
```

Comment:
Are we putting a deprecation date on this pathway, or is it going to have to live forever? Right now it's on the read side, so to get rid of it we'd have to automigrate users' skins somehow I imagine.

Reply:
I guess we'll do this one day. I'll note this in the inline comments.